lib/gossip/dissemination.js

```diff
@@ -156,6 +156,9 @@ Dissemination.prototype._issueAs = function _issueAs(filterChange, mapChanges) {
 
         if (change.piggybackCount > this.maxPiggybackCount) {
             delete this.changes[address];
+            if (Object.keys(this.changes).length === 0) {
```

Comment:
can you add a `lib/gossip/events.js` file and define a ChangesExhaustedEvent therein (take a look at lib/ring/events.js) for an example. Recently, I've been trying to be a little bit more explicit about the events that Ringpop emits in its various modules.

then this file'd look like:

```
this.emit('changesExhaused', new events.ChangesExhaustedEvent());
```